packages/trader/src/Modules/Contract/Components/LastDigitPrediction/last-digit-prediction.tsx

```diff
@@ -79,11 +79,11 @@ const LastDigitPrediction = ({
             [key: string]: (val: number | null) => boolean;
         } = {
             DIGITMATCH: (val: number | null) => val === barrier,
-            DIGITDIFF: (val: number | null) => val !== barrier && !!barrier && !isNaN(barrier),
-            DIGITOVER: (val: number | null) => !!(val && barrier) && val > barrier,
-            DIGITUNDER: (val: number | null) => !!(val && barrier) && val < barrier,
+            DIGITDIFF: (val: number | null) => val !== barrier,
+            DIGITOVER: (val: number | null) => !!((val || val === 0) && (barrier || barrier === 0)) && val > barrier,
```

Comment:
We can also check `typeof`:
` DIGITOVER: (val: number | null) => (typeof val  === number) && (typeof barrier === number) && val > barrier`
Not sure if it's better, please, ignore if not relevant

Reply:
Well, both solutions are okay, I prefer the shorter one though 😄